code/modules/atmospherics/machinery/components/fusion/hfr_main_processes.dm

```diff
@@ -357,142 +194,172 @@
-						internal_output.assert_gases(/datum/gas/zauker, /datum/gas/stimulum)
-						internal_output.gases[/datum/gas/zauker][MOLES] += scaled_production * 5.35
-						internal_output.gases[/datum/gas/stimulum][MOLES] += scaled_production * 2.15
-						moderator_internal.gases[/datum/gas/proto_nitrate][MOLES] -= min(moderator_internal.gases[/datum/gas/proto_nitrate][MOLES], scaled_production * 3.35)
-						radiation *= 2
-						heat_output *= 2.25
-					if(moderator_list[/datum/gas/bz])
-						for(var/mob/living/carbon/human/human in view(src, HALLUCINATION_HFR(heat_output)))
-							//mesons won't protect you at fusion level 6
-							var/distance_root = sqrt(1 / max(1, get_dist(human, src)))
-							human.hallucination += power_level * 150 * distance_root
-							human.hallucination = clamp(human.hallucination, 0, 200)
-						moderator_internal.gases[/datum/gas/antinoblium][MOLES] += clamp((scaled_fuel_list[scaled_fuel_list[3]] / (fuel_injection_rate * 0.045)), 0, 10) * delta_time
-					if(moderator_list[/datum/gas/healium] > 100)
-						if(critical_threshold_proximity > 400)
-							critical_threshold_proximity = max(critical_threshold_proximity - (moderator_list[/datum/gas/healium] / 100 * delta_time ), 0)
-							moderator_internal.gases[/datum/gas/healium][MOLES] -= min(moderator_internal.gases[/datum/gas/healium][MOLES], scaled_production * 20)
-					internal_fusion.gases[/datum/gas/antinoblium][MOLES] += 0.01 * (scaled_fuel_list[scaled_fuel_list[3]] / (fuel_injection_rate * 0.095)) * delta_time
+	// Run the common effects, committing changes where applicable
+
+	// This is repetition, but is here as a placeholder for what will need to be done to allow concurrently running multiple recipes
+	var/common_production_amount = production_amount * selected_fuel.gas_production_multiplier
+	moderator_common_process(delta_time, common_production_amount, internal_output, moderator_list, dirty_production_rate, heat_output, radiation_modifier)
+
+/**
+ * Perform recipe specific actions. Fuel consumption and recipe based gas production happens here.
+ */
+/obj/machinery/atmospherics/components/unary/hypertorus/core/proc/moderator_fuel_process(delta_time, production_amount, consumption_amount, datum/gas_mixture/internal_output, moderator_list, datum/hfr_fuel/fuel, fuel_list)
+	// Adjust fusion consumption/production based on this recipe's characteristics
+	var/fuel_consumption = consumption_amount * 0.85 * selected_fuel.fuel_consumption_multiplier
+	var/scaled_production = production_amount * selected_fuel.fuel_consumption_multiplier
+
+	for(var/gas_id in fuel.requirements)
+		internal_fusion.gases[gas_id][MOLES] -= min(fuel_list[gas_id], fuel_consumption)
+	for(var/gas_id in fuel.primary_products)
+		internal_fusion.gases[gas_id][MOLES] += fuel_consumption * 0.5
+
+	// Each recipe provides a tier list of six output gases.
+	// Which gases are produced depend on what the fusion level is.
+	var/list/tier = fuel.secondary_products
+	switch(power_level)
```

Comment:
I wish there was a better way to do this

Reply:
Yeah. IIRC there was dynamic behavior coming later down the pipeline, can probably make things better here once we know what that looks like

Reply:
That would be really nice

Reply:
Mapped out what we currently have:
![hfr-prod](https://user-images.githubusercontent.com/14355175/134131701-a0f42d79-bd7d-430b-a42f-da20fceabd8d.png)
Yeah, communicating that to players is going to be pretty difficult.

Rough thoughts:
 - Mapping fusion power level to output tier level is a headache. It's probably important to preserve that output tier levels 5 and 6 should mostly be from the dangerous fusion power levels 5 and 6. Maybe make "plasma gets you a low-rate preview of the next level" a rule rather than a weird exception at fusion power level 2, so the top-right mapping looks like this:
   ![hfr-prod-proposed](https://user-images.githubusercontent.com/14355175/134132630-e7388cd6-2474-4ffb-9f20-5d8d2a2c20e3.png)
   and change coefficients to operate off some curve which has production rates peak at fusion power level 4, rather than hardcoded variables
 - The output tier of Nitrogen and Plasma should probably be swapped so Plasma is the higher tier
 - Stimulum is probably a bit too easy to get to as well?
 - The common effects are also going to be pretty hard to communicate, though there are commonalities. Making Fusion Level 6 be a "weird shit happens here" level seems fine, given how nuts things get if you reach there. Giving the three moderator gases the same progression shape, removing gaps, and making distinct lists for each gives the following all up:
![hfr-prod-proposed-2](https://user-images.githubusercontent.com/14355175/134136946-b00f709a-8a3d-4489-83f7-57380f788a5e.png)
   or in progression map form:
![hfr-prod-proposed-3](https://user-images.githubusercontent.com/14355175/134138647-b8c12e12-8178-4cca-a405-4d5720a5a27e.png)
   ...that's getting well into "separate feature PR" territory, though

@Ghilker thoughts? Would this sort of thing get in the way of dynamic effects?

Reply:
i really like the progression map form, i don't think it will get in the way of anything, but it will enhance the HFR experience by bringing a sort of "progression system" on the table. tho yes this is separate feature PR territory